### PR TITLE
Add desktop notifications

### PR DESCRIPTION
**Pull request by antonlindstrom**
This adds desktop notifications for Gone to be able to see when a
session has ended. The change introduces a message that will say for
instance that "Your working session has ended, time for a short break".
A Notifier interface has been added to make sure that the feature can be
extended with custom notifiers.

Desktop notifications is nice to have to be able to get an interruption
when a session is done.

Thanks!

### CODE REVIEW

File: state/state.go [hunk 1]
```diff
@@ -99,7 +99,22 @@ func (s *State) Message() string {
...
+		'w': "working session",
+		's': "short break",
+		'l': "long break",
+	}
+
+	if !s.IsEnded() {
+		return fmt.Sprintf("A %s is in progress", msgs[s.pattern[s.currentIdx]])
+	}
+
+	return fmt.Sprintf("Your %s has ended, time for a %s.", msgs[s.pattern[s.currentIdx]], msgs[s.pattern[s.currentIdx+1]])
```
Comment (cwarden) [inline state/state.go]: We need to get `msgs[s.pattern[0]]` for the next message if `s.currentIdx == len(s.pattern)-1`.

Comment (antonlindstrom) [reply state/state.go]: Thank you for the feedback and sorry for the delay in fixing it. The latest commit should have the changes requested as well as validation that it hopefully works. :bow:

**Comment by antonlindstrom** [comment]
@guillaumebreton Any updates on this? Would you consider this change?

Thanks!

**Review by guillaumebreton — APPROVED** [review]
Thanks for the PR 😊 and sorry for the delayed review

**Comment by antonlindstrom** [comment]
Awesome, thank you so much for this! :bowing_man: 

Have an amazing day!

**Pull request merged**